xword.py

```diff
@@ -2,20 +2,48 @@
 # -*- coding: utf-8 -*-
 """Crossword Solver Program"""
 
-__author__ = "???"
-
-# YOUR HELPER FUNCTION GOES HERE
+__author__ = """Kyle Thomas, with the use of:
+             https://regex101.com/
+             https://docs.python.org/3/library/re.html
+             """
+
+import re
+
+
+def make_regex(word):
+    "Creates a regex pattern based on the test_word input"
+    word_regex = r'^'
+    for char in word:
+        if char == ' ':
+            word_regex += r'\w'
+        else:
+            word_regex += char
+    word_regex += r'$'
+    return word_regex
+
+
+def find_matches(word_list, word, pattern):
+    "Searches word_list for possible matches"
+    matches = []
+    for item in word_list:
+        if len(item) == len(word) and re.search(pattern, item):
```

Comment:
instead of comparing lengths here, you could update your regex pattern to make sure the pattern matches the full string. Reference: https://stackoverflow.com/questions/5997673/regex-to-match-full-string

Reply:
Thanks - I was already following that pattern in my make_regex function, so I just refactored the find_matches function to take two arguments and updated the function call in main.